src/main/resources/assets/admin/common/js/data/PropertySet.ts

```diff
@@ -461,6 +461,38 @@ module api.data {
             return true;
         }
 
+        public syncEmptyArrays(target: PropertySet) {
+
+            if(!target) {
+                return;
+            }
+
+            target.getPropertyArrays().forEach((propertyArray: PropertyArray) => {
+                if (!this.getPropertyArray(propertyArray.getName())) {
+
+                    this.addPropertyArray(propertyArray.copy(this));
+                }
+
+                propertyArray.forEach(((property, index) => {
+                    if (property.getType().equals(api.data.ValueTypes.DATA)) {
+
+                        const targetPropertySet = property.getValue().getPropertySet();
+
+                        const thisPropertyArray = this.getPropertyArray(propertyArray.getName());
+                        const thisProperty = thisPropertyArray.get(index);
+                        if (thisProperty && thisProperty.getType().equals(api.data.ValueTypes.DATA)) {
+
+                            const thisPropertySet = thisProperty.getValue().getPropertySet();
+
```

Comment:
same here - ``thisProperty.getPropertySet()``

Actually, instead of doing 
```
const thisProperty = thisPropertyArray.get(index);
const thisPropertySet = thisProperty.getValue().getPropertySet();
```
you could simply do
```
const thisPropertySet = thisPropertyArray.getSet(index);
```